dashboard/client/src/pages/serve/ServeApplicationRow.tsx

```diff
@@ -27,44 +51,82 @@ export const ServeApplicationRow = ({
-      </TableCell>
-      <TableCell align="center">
-        {message ? (
-          <CodeDialogButtonWithPreview title="Message details" code={message} />
-        ) : (
-          "-"
-        )}
-      </TableCell>
-      <TableCell align="center">{Object.values(deployments).length}</TableCell>
-      <TableCell align="center">
-        {formatDateFromTimeMs(last_deployed_time_s * 1000)}
-      </TableCell>
-      <TableCell align="center">
-        <DurationText startTime={last_deployed_time_s * 1000} />
-      </TableCell>
-      <TableCell align="center">
-        {deployed_app_config ? (
-          <CodeDialogButton
-            title={
-              name ? `Application config for ${name}` : `Application config`
-            }
-            code={deployed_app_config}
+    <React.Fragment>
+      <TableRow>
+        <TableCell>
+          <IconButton size="small" onClick={onExpandButtonClick}>
+            {!isExpanded ? (
+              <RiArrowRightSLine className={classes.expandCollapseIcon} />
+            ) : (
+              <RiArrowDownSLine className={classes.expandCollapseIcon} />
+            )}
+          </IconButton>
+        </TableCell>
+        <TableCell align="center" className={classes.applicationName}>
+          <Link
+            component={RouterLink}
+            to={`applications/${name ? encodeURIComponent(name) : "-"}`}
+          >
+            {name ? name : "-"}
+          </Link>
+        </TableCell>
+        <TableCell align="center">
+          <StatusChip type="serveApplication" status={status} />
+        </TableCell>
+        <TableCell align="center">
+          {message ? (
+            <CodeDialogButtonWithPreview
+              title="Message details"
+              code={message}
+            />
+          ) : (
+            "-"
+          )}
+        </TableCell>
+        <TableCell align="center">
+          {/* placeholder for num_replicas, which does not apply to an application */}
+          -
+        </TableCell>
+        <TableCell align="center">
+          {deployed_app_config ? (
+            <CodeDialogButton
+              title={
+                name ? `Application config for ${name}` : `Application config`
+              }
+              code={deployed_app_config}
+              buttonText="View config"
+            />
+          ) : (
+            "-"
+          )}
+        </TableCell>
+        <TableCell align="center">{route_prefix}</TableCell>
+        <TableCell align="center">
+          {formatDateFromTimeMs(last_deployed_time_s * 1000)}
+        </TableCell>
+        <TableCell align="center">
+          <DurationText startTime={last_deployed_time_s * 1000} />
+        </TableCell>
+      </TableRow>
+      {isExpanded &&
+        deploymentsList.map((deployment) => (
+          <ServeDeploymentRow
+            key={`${application.name}-${deployment.name}`}
+            deployment={deployment}
+            application={application}
+            showExpandColumn={true}
```

Comment:
nit: with react components,  instead of sending in `={true}` you can just do `<Component showExpandColumn />`.

This is also the reason why dropping the `showExpandColumn` should default the value to false.

Reply:
Thanks, good to know! Fixed bd1b086